### ISSUE TITLE
[bug]: seam painting regression

### ISSUE THREAD

**Issue by psychedelicious** [invoke-ai/InvokeAI#3361]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Linux

### GPU

cuda

### What version did you experience this issue on?

feat/ui/nodes-canvas

### What happened?

Seam painting looking kinda fried all the time. I'm not sure what is going on, but my settings are being used. Cannot find a combination of seam settings, sampler, etc that produce results as good as 2.3.

You can use the `feat/ui/nodes-canvas` branch to test, though it's still a WIP.

### Screenshots

![image](https://user-images.githubusercontent.com/4822129/236846027-4fd2cb52-b90f-48cb-92ee-51cdfdd94425.png)
![image](https://user-images.githubusercontent.com/4822129/236846175-8e1bb65b-461e-40f6-8788-3eb4c50896b8.png)
![image](https://user-images.githubusercontent.com/4822129/236846209-9c3681f0-4158-452a-8efd-c0c14d20295b.png)
![image](https://user-images.githubusercontent.com/4822129/236846238-a3728e6a-7a2c-4908-9d7b-289bff176d26.png)

**Comment by psychedelicious** [comment]
@Kyle0654 Would you mind having a look? 

In this branch, the canvas uses the existing Inpainting node. I suspect something is not quite right. Also in this branch, I've made Infill nodes for each Infill type, and done a bit of simple cleanup on several nodes - but the canvas does not use that stuff.

**Comment by psychedelicious** [comment]
@Kyle0654 Yep, parameters are there and the output varies if I change them.

I forcibly enabled image debugging:
`invokeai/backend/generator/inpaint.py`:
```python
class Inpaint(Img2Img):
    def __init__(self, model, precision):
        self.inpaint_height = 0
        self.inpaint_width = 0
        self.enable_image_debugging = True <-- forcibly enable it
```

The `init_filled` image (infilled with patchmatch) looks great, but it looks like the incorrect area is masked? This is the first invocation of the inpaint function:
![debug](https://user-images.githubusercontent.com/4822129/236988798-5e6cf6e6-d6ff-4626-8b1e-530cb7358f25.png)

I'm a bit lost with how tangly this code is - the generation functions are methods on stateful classes which mutate their own attributes (see ~L360 `inpaint.py`). I can't really make sense of it. Hoping it makes sense to you.

**Comment by Kyle0654** [comment]
Yah this is why I've been advocating we just move to a nodes-based approach.

**Comment by psychedelicious** [comment]
I've attempted to do that but it's a bit too tangle-y. I made infill nodes in #3323 but that's as far as I've gotten. Appreciate help nodifying it!

**Comment by psychedelicious** [comment]
fixed in #3449
